apps/settings/js/panels/operator_settings/model/auto_selection_model.js

```diff
@@ -0,0 +1,93 @@
+define(function() {
+  'use strict';
+
+  var Defer = require('modules/defer');
+  var FeatureModel = require('modules/feature_model');
+  var MobileConnectionWrapper =
+    require('panels/operator_settings/mobile_connection_wrapper');
+
+  // Auto selection state
+  const AS_STATE = {
+    UNKNOWN: 0,
+    ENABLED: 1,
+    DISABLED: 2
+  };
+
+  const AS_STATE_MAP = {
+    'manual': AS_STATE.DISABLED,
+    'automatic': AS_STATE.ENABLED
+  };
+
+  var AutoSelectionModel = function(connWrapper) {
+    return FeatureModel({
+      onInit: function() {
+        this._pendingRequest = null;
+        connWrapper.observe('state', (newState) => {
+          this._checkPendingRequest(newState);
+        });
+        return AS_STATE_MAP[connWrapper.networkSelectionMode] ||
+          AS_STATE.UNKNOWN;
+      },
+      onGetState: function() {
+        return Promise.resolve(AS_STATE_MAP[connWrapper.networkSelectionMode] ||
+          AS_STATE.UNKNOWN);
+      },
+      onSetState: function(state) {
+        switch (connWrapper.state) {
+          case MobileConnectionWrapper.STATE.IDLE:
+            if (state === AS_STATE.ENABLED) {
+              return connWrapper.setAutoSelection().then(() => {
+                return AS_STATE.ENABLED;
+              }, () => {
+                return AS_STATE.DISABLED;
+              });
+            } else {
+              return Promise.resolve(AS_STATE.DISABLED);
+            }
+            break;
```

Comment:
we will definitely return something here in `IDLE` state, so this `break;` may be redundant.